workspaces/ui-v2/src/optic-components/hooks/useShapeDescriptor.ts

```diff
@@ -145,7 +145,7 @@ export function useShapeDescriptor(
               })
             );
             choice.asObject.fields = newFields;
-            return choice;
+            return { ...choice, shapeId: choice.id };
```

Comment:
These are the only changes we need to do in code - which is related to the https://github.com/opticdev/optic/pull/807 PR (i.e. shim id -> shapeId)